Oci 5.1.0 japaccoe 2851 release 5.1.0

ocisunbird opened the pull request:
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

heungheung reviewed (changes requested):
request changes for
- ansible/roles/es6/templates/elasticsearch.yml.j2
- ansible/roles/es6/templates/jvm.options.j2
- ansible/roles/log-es6/templates/jvm.options.j2
as this will apply to ALL 

IMHO
1. this should only applies to OCI
2. the debug options should not be included as default installation

heungheung commented:
@simhead please take a look

ocisunbird commented:
> request changes for
> 
>     * ansible/roles/es6/templates/elasticsearch.yml.j2
> 
>     * ansible/roles/es6/templates/jvm.options.j2
> 
>     * ansible/roles/log-es6/templates/jvm.options.j2
>       as this will apply to ALL
> 
> 
> IMHO
> 
>     1. this should only applies to OCI
> 
>     2. the debug options should not be included as default installation

removed debug options - please check and confirm

ddevadat commented:
@simhead 

Use the below sample to create an if else based template

We are setting cloud_service_provider as an ansible variable.

```
{% if cloud_service_provider == "oci" %}
key=value
{% else %}
some_other_key=value
{% endif %}
```

ocisunbird commented:
@heungheung 
1. this should only applies to OCI
- added "oci" condition for 2 impacted files: i.e. es6 and log-es6 jvm.options.j2.
2. the debug options should not be included as default installation
- removed debug options

heungheung reviewed (changes requested):
sorry.... missed 1 entry in my previous review
  ansible/roles/es-oci-snapshot/defaults/main.yml
why endpoint got hardcoded?

ocisunbird commented:
@heungheung , oh..sorry. removed these hard coded values. please re-check.

heungheung reviewed (approved):
good to go
